src/main/java/net/snowflake/client/jdbc/cloud/storage/GCSAccessStrategyAwsSdk.java

```
@@ -0,0 +1,302 @@
+  private final AmazonS3 amazonClient;
+
+  GCSAccessStrategyAwsSdk(StageInfo stage) {
+    String accessToken = (String) stage.getCredentials().get("GCS_ACCESS_TOKEN");
+
+    // todo: fix client configuration needed
+    /*
+    clientConfig.getApacheHttpClientConfig().setSslSocketFactory(getSSLConnectionSocketFactory());
+    if (session != null) {
+        S3HttpUtil.setProxyForS3(session.getHttpClientKey(), clientConfig);
+    } else {
+        S3HttpUtil.setSessionlessProxyForS3(proxyProperties, clientConfig);
+    }*/
+
+    Optional<String> oEndpoint = stage.gcsCustomEndpoint();
+    String endpoint = "storage.googleapis.com";
+    if (oEndpoint.isPresent()) {
+      endpoint = oEndpoint.get();
+    }
+    if (endpoint.startsWith("https://")) {
+      endpoint = endpoint.replaceFirst("https://", "");
+    }
+    if (endpoint.startsWith(stage.getStorageAccount())) {
+      endpoint = endpoint.replaceFirst(stage.getStorageAccount() + ".", "");
+    }
+
+    AmazonS3ClientBuilder amazonS3Builder =
+        AmazonS3Client.builder()
+            .withPathStyleAccessEnabled(false)
+            .withEndpointConfiguration(
+                new AwsClientBuilder.EndpointConfiguration(endpoint, "auto"));
+
+    ClientConfiguration clientConfig = new ClientConfiguration();
+
+    SignerFactory.registerSigner(
+        "net.snowflake.client.jdbc.cloud.storage.AwsSdkGCPSigner",
+        net.snowflake.client.jdbc.cloud.storage.AwsSdkGCPSigner.class);
+    clientConfig.setSignerOverride("net.snowflake.client.jdbc.cloud.storage.AwsSdkGCPSigner");
+
+    if (accessToken != null) {
+      amazonS3Builder.withCredentials(
+          new AWSStaticCredentialsProvider(new BasicAWSCredentials(accessToken, "")));
+    } else {
+      logger.debug("no credentials provided, configuring bucket client without credentials");
+      amazonS3Builder.withCredentials(
+          new AWSStaticCredentialsProvider(new BasicAWSCredentials("", "")));
+    }
```

I don't think this code branch can ever succeed. Can we throw an exception instead?

---

It was implemented for the GCS part. Not sure how to trigger it